Automattic/jetpack
# CRM: Resolves 3091 - clean up user roles

Resolves Automattic/zero-bs-crm#3091 - clean up user roles

## Proposed changes:

In the linked issue, a user cited our docs stating that the Mail Manager should not be able to edit quotes or invoices:

https://kb.jetpackcrm.com/knowledge-base/setting-up-your-team/

Looking at the roles, I confirmed that confirmed that we were lax in most of our permissions; most roles could view and edit most objects. I cleaned up the code and tried to standardise the role definitions as follows:

* CRM Admin: full access to everything
* WP Admin: full access to everything
* Contact Manager: view and edit all objects, but not settings
* Quote Manager: view and edit contacts and quotes
* Invoice Manager: view and edit contacts, invoices, and transactions (as transactions are directly linked to invoices)
* Transaction Manager: view and edit contacts and transactions
* CRM Customer: custom role with no special capabilities (used for Client Portal)
* Mail Manager: view all objects (in listview), and edit contacts and tasks

Note that all team member roles can edit contacts, as that permission is deeply engrained in the system checks that are not directly related to contacts.

Prior to this PR, anyone could add a new object (even if they could edit it). This adds a basic check to prevent that (it's not pretty at this stage, but no one should run into it accidentally).

I also restricted display of the Dashboard and Task Scheduler according to their proper permissions.

Finally, I co-opted an old migration so that the roles would be properly refreshed (deleted and added) in WordPress.

I have not updated the KB to reflect the above, but we'll need to do so if this gets merged.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Essentially make a user and switch its role (I used a private browser) to make sure things are properly restricted per my outline above.
